client/app/src/services/network.service.js

```diff
@@ -348,22 +347,32 @@
 
     // Updates peer with all currency values relative to the USD price.
     function updatePeerWithCurrencies(peer, res) {
-      $http.get('https://api.fixer.io/latest?base=USD', { timeout: 2000}).then( function (result) {
-        const USD_PRICE = Number(res.data[0].price_usd)
-        var currencies = ["aud", "brl", "cad", "chf", "cny", "eur", "gbp", "hkd", "idr", "inr", "jpy", "krw", "mxn", "rub"]
+      var currencies = ["aud", "brl", "cad", "chf", "cny", "eur", "gbp", "hkd", "idr", "inr", "jpy", "krw", "mxn", "rub", "usd"]
+      var currency_request_url = createCurrencyApiCall(currencies)
+      $http.get(currency_request_url, { timeout: 2000}).then( function (result) {
+        const EUR_PRICE = Number(res.data[0].price_eur)
         var prices = {}
         currencies.forEach(function(currency) {
-          prices[currency] = result.data.rates[currency.toUpperCase()] * USD_PRICE
+          prices[currency] = result.data.rates[currency.toUpperCase()] * EUR_PRICE
         })
-        prices["btc"] = res.data[0].price_btc
-        prices["usd"] = res.data[0].price_usd
+        prices["btc"] = Number(res.data[0].price_btc)
+        prices["eur"] = Number(EUR_PRICE)
         peer.market.price = prices
         storageService.setGlobal('peerCurrencies', prices)
       })
 
       return peer
     }
 
+    function createCurrencyApiCall(currencies) {
+        var get_request = 'https://api.fixer.io/latest?symbols='
+        currencies.forEach(function(currency) {
+            get_request += '' + currency.toUpperCase() + ','
```

Comment:
A template literal would be a bit cleaner here

Reply:
Cleaned it up by using .toString on the currencies array instead, should have done that initially.